developer_manual/basics/dependency_injection.rst

```diff
@@ -310,45 +310,52 @@ The following parameter names and type hints can be used to inject core services
 Parameters:
 
 * **AppName**: The app id
-* **WebRoot**: The path to the Nextcloud installation
 * **UserId**: The id of the current user
+* **WebRoot**: The path to the Nextcloud installation
+
+Aliases:
+* **appName**: resolves to ``AppName``
+* **Request**: resolves to ``\OCP\\IRequest``
+* **ServerContainer**: resolves to ``\OCP\IServerContainer``
+* **userId**: resolves to ``UserId``
+* **webRoot**: resolves to ``WebRoot``
 
 Types:
 
-* **OCP\\IAppConfig**
-* **OCP\\IAppManager**
-* **OCP\\IAvatarManager**
-* **OCP\\Activity\\IManager**
-* **OCP\\ICache**
-* **OCP\\ICacheFactory**
-* **OCP\\IConfig**
-* **OCP\\AppFramework\\Utility\\IControllerMethodReflector**
-* **OCP\\Contacts\\IManager**
-* **OCP\\IDateTimeZone**
-* **OCP\\IDBConnection**
-* **OCP\\Diagnostics\\IEventLogger**
-* **OCP\\Diagnostics\\IQueryLogger**
-* **OCP\\Files\\Config\\IMountProviderCollection**
-* **OCP\\Files\\IRootFolder**
-* **OCP\\IGroupManager**
-* **OCP\\IL10N**
-* **OCP\\ILogger**
-* **OCP\\BackgroundJob\\IJobList**
-* **OCP\\INavigationManager**
-* **OCP\\IPreview**
-* **OCP\\IRequest**
-* **OCP\\AppFramework\\Utility\\ITimeFactory**
-* **OCP\\ITagManager**
-* **OCP\\ITempManager**
-* **OCP\\Route\\IRouter**
-* **OCP\\ISearch**
-* **OCP\\ISearch**
-* **OCP\\Security\\ICrypto**
-* **OCP\\Security\\IHasher**
-* **OCP\\Security\\ISecureRandom**
-* **OCP\\IURLGenerator**
-* **OCP\\IUserManager**
-* **OCP\\IUserSession**
+* ``\OCP\IAppConfig``
+* ``\OCP\IAppManager``
+* ``\OCP\IAvatarManager``
+* ``\OCP\Activity\IManager``
+* ``\OCP\ICache``
+* ``\OCP\ICacheFactory``
+* ``\OCP\IConfig``
+* ``\OCP\AppFramework\Utility\IControllerMethodReflector``
+* ``\OCP\Contacts\IManager``
+* ``\OCP\IDateTimeZone``
+* ``\OCP\IDBConnection``
+* ``\OCP\Diagnostics\IEventLogger``
+* ``\OCP\Diagnostics\IQueryLogger``
+* ``\OCP\Files\Config\IMountProviderCollection``
+* ``\OCP\Files\IRootFolder``
+* ``\OCP\IGroupManager``
+* ``\OCP\IL10N``
+* ``\OCP\ILogger``
```

Comment:
Should we rather list the LoggerInterface here?

Reply:
Yes, makes sense. And also mark the ILogger as deprecated.

I did not check this list. There is a lot more that can be injected these days and I don't know how to get a reliable list for that info. This has to do with auto-wiring. In the past it was easier to see what is registered on ever app container.

Reply:
Right, getting a full list is probably hard